src/search/videoMetadata.ts

```diff
@@ -0,0 +1,35 @@
+import getVideosId from "./videoId.ts"
+import client from "../api/client.ts"
+import { Video } from "../models/Video.ts"
+import { VideoSchema } from "../schema/VideoSchema.ts"
+
+async function getVideosMetadata(): Promise<Video[]> {
+    
+    try {
+        const videosIdArray = await getVideosId()
+        const response = await client.videos.list({
+            part: ["snippet,statistics"],
+            id: videosIdArray,
+        })
+        if (response.data.items) {
+            const videosMetadata = response.data.items.map(
+                ({ id, snippet, statistics }) => ({
+                    videoId: id,
+                    publishedAt: snippet?.publishedAt,
+                    likeCount: statistics?.likeCount,
+                    viewCount: statistics?.viewCount,
+                })
+            )
+            const validatedVideos = videosMetadata.map((video) =>
+                VideoSchema.parse(video)
```

Comment:
Please use .safeParse()

Reply:
After reviewing the Zod documentation, I notice that safeParse returns an object in the format of `{ success: true; data: T; } | { success: false; error: ZodError; } `in order to handle errors. This has caused some complications by restructuring the code. Thus, I decided to utilize `.parse()` within try and catch blocks, providing similar functionality as safeParse but in a extended manner.

```
let validatedVideos;
      try {
        validatedVideos = videosMetadata.map((video) =>
          VideoSchema.parse(video)
        );
      } catch (error) {
        throw new Error("Error parsing video metadata with zod");
      }

      return validatedVideos;
```
      
What are your opinions about that?